example/tutorials/t8_step1_coarsemesh.c

```diff
@@ -0,0 +1,127 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/* In this example we build a coarse mesh with a cube geometry.
+ * The cube is meshed with 6 coarse tetrahedra.
+ * We then output it in vtu format and destroy it.
+ * 
+ * How you can experiment here:
+ *  - Use Paraview to visualize the output files.
+ *  - Change the parameters of t8_cmesh_new_hypercube
+ *    You can change the element shape or switch to a partitioned cmesh or use
+ *    periodic boundaries.
+ *  - Exchange t8_cmesh_new_hypercube with any other t8_cmesh_new function
+ *    from t8_cmesh.h to create a different cmesh.
+ */
+
+#include <t8.h>
+#include <t8_cmesh.h>
+#include <t8_cmesh_vtk.h>
+
+/* Builds cmesh of 6 tetrahedra that build up a unit cube.
+ * \param [in] comm   MPI Communicator to use.
+ * \return            The coarse mesh.
+ */
+static t8_cmesh_t
+t8_step1_build_tetcube_coarse_mesh (sc_MPI_Comm comm)
+{
+  t8_cmesh_t          cmesh;
+
+  /* Build a coarse mesh of 6 tetrahedral trees that form a cube.
+   * You can modify the first parameter to build a cube with different
+   * tree shapes, i.e. T8_ECLASS_QUAD for a unit square with 1 quadrilateral tree.
+   * See t8_eclass.h, t8_cmesh.h for all possible shapes.
+   * 
+   * The second argument is the MPI communicator to use for this cmesh.
+   * The remaining arguments are 3 flags that control
+   *   do_bcast     - If non-zero only the root process will build the cmesh and will broadcast it to the other processes. The result is the same.
+   *   do_partition - If non-zero the cmesh will be partitioned among the processes. If 0 each process has a copy of the whole cmesh.
+   *   periodic     - If non-zero the cube will have periodic boundaries. That is, i.e. the left face is connected to the right face.
+   */
+  cmesh = t8_cmesh_new_hypercube (T8_ECLASS_TET, comm, 0, 0, 0);
```

Comment:
raises Assertion with  cmesh = t8_cmesh_new_hypercube (T8_ECLASS_TET, comm, 0, 1, 0);

Reply:
Damn, this seems to be a major bug. See #79 and #80.
Would shall we do? Wait until the bug is resolved?
Or merge this regardless, since the bug is in develop nevertheless?

Edit: Okay for now i added a warning that is printed when the hypercube cmesh is created.
Do you believe this is a viable solution until the cmesh bug is resolved?

Reply:
Since you opened issues regarding this bug, I think it is a viable solution.